datafusion-cli/src/exec.rs

```diff
@@ -221,37 +222,73 @@ async fn exec_and_print(
 
     let statements = DFParser::parse_sql_with_dialect(&sql, dialect.as_ref())?;
     for statement in statements {
+        let adjusted =
```

Comment:
I extracted the existing code to update the print options based on statement into its own structuer